docs/book/src/clusterctl/provider-contract.md

```diff
@@ -37,6 +37,60 @@ It is possible to customize the list of providers for `clusterctl` by changing t
 
 </aside>
 
+#### Adding a provider to clusterctl
+
+As a Cluster API project, we always have been more than happy to give visibility to all the open source CAPI providers
+by allowing provider's maintainers to add their own project to the pre-defined list of provider shipped with `clusterctl`.
+
+<aside class="note">
+
+<h1>Important! it is visibility only</h1>
+
+Provider's maintainer are the ultimately responsible for their own project.
+
+Adding a provider to the `clusterctl` provider list does not imply any form of quality assessment, market screening, 
+entitlement, recognition or support by the Cluster API maintainers.
+
+</aside>
+
+This is the process to add a new provider to the pre-defined list of providers shipped with `clusterctl`:
+- As soon as possible, create an issue to the [Cluster API repository](https://sigs.k8s.io/cluster-api) declaring the intent to add a new provider;
+  each provider must have a unique name & type in the pre-defined list of providers shipped with `clusterctl`; the provider's name
+  must be declared in the issue above and abide to the following naming convention:
+  - The name must consist of lower case alphanumeric characters or '-', and must start and end with an alphanumeric character.
+  - The name length should not exceed 63 characters.
+  - For providers not in the kubernetes-sigs org, in order to prevent conflicts the `clusterctl` name must be prefixed with
+    the provider's GitHub org name followed by `-` (see note below).
+- Create a PR making the necessary changes to clusterctl and the Cluster API book, e.g. [#9798](https://github.com/kubernetes-sigs/cluster-api/pull/9798),
+  [9720](https://github.com/kubernetes-sigs/cluster-api/pull/9720/files). 
+
+The Cluster API maintainers will review issues/PRs for adding new providers. If the PR merges before code freeze deadline
+for the next Cluster API minor release, changes will be included in the release, otherwise in the next minor
+release. Maintainers will also consider if possible/convenient to backport to the current Cluster API minor release
+branch to include it in the next patch release.
+
+<aside class="note">
+
+<h1>What about closed source providers?</h1>
+
+Closed source provider can not be added to the pre-defined list of provider shipped with `clusterctl`, however, 
+those providers could be used with `clusterctl` by changing the [clusterctl configuration](configuration.md).
+
+</aside>
+
+<aside class="note">
+
+<h1>Provider's GitHub org prefix</h1>
+
+The need to add a prefix for providers not in the kubernetes-sigs org applies to all the providers being added to
+`clusterctl`'s pre-defined list of provider starting from January 2024. This rule doesn't apply retroactively
+to the existing pre-defined providers, but we reserve the right to reconsider this in the future.
```

Comment:
Something to think about, and potentially to discuss at next week's meeting is that we have added all providers to clusterctl so far without much restriction. While this has been generally fine, there are also consideration regarding license restrictions, or CVEs that we might carry unexpectedly by allowing the blind installation of providers that might look like they're part of the SIG.

Maybe it's time for a breaking change here?

Reply:
I will bring this up at office hours; the tricky part is how to enforce those rules over time without adding more payload to the core maintainers

Reply:
I think we don't even have continuous license scanning implemented in core CAPI

Reply:
see https://github.com/kubernetes-sigs/cluster-api/pull/9975#issuecomment-1896390601